### PR TITLE
core(meta-elements): add NodeDetails

### LINKED ISSUE #12263: use NodeDetails in MetaElements
I removed this from #11877 but never followed up.

https://github.com/GoogleChrome/lighthouse/pull/11877#issuecomment-750417392

code: https://github.com/GoogleChrome/lighthouse/pull/11877/commits/cc8cdb375facd623b258a37758afbec383627dc9

### PR DESCRIPTION
**Pull request by connorjclark**
Fixes #12263

### CODE REVIEW

File: types/artifacts.d.ts [hunk 1]
```diff
@@ -115,7 +115,7 @@ declare global {
       /** All the link elements on the page or equivalently declared in `Link` headers. @see https://html.spec.whatwg.org/multipage/links.html */
       LinkElements: Artifacts.LinkElement[];
       /** The values of the <meta> elements in the head. */
-      MetaElements: Array<{name?: string, content?: string, property?: string, httpEquiv?: string, charset?: string}>;
+      MetaElements: Array<{name?: string, content?: string, property?: string, httpEquiv?: string, charset?: string, node: LH.Artifacts.NodeDetails}>;
```
Comment (connorjclark) [inline types/artifacts.d.ts:118]: This change introduces a type error on a seemingly unrelated file (`stacks.js` snapshot fn)

File: lighthouse-core/test/results/artifacts/artifacts.json [hunk 2]
```diff
@@ -3220,4 +3266,4 @@
       "columnNumber": 36
     }
   ]
-}
```
Comment (adamraine) [inline lighthouse-core/test/results/artifacts/artifacts.json:3223]: nit: newline

File: types/artifacts.d.ts (hunk 1 above)
Comment (adamraine) [reply types/artifacts.d.ts:118]: I'm having a hard time finding a link as well. Could this be a TS bug?

I was able to fix the error by making `node` optional though.

Comment (brendankenny) [reply types/artifacts.d.ts:118]: https://github.com/GoogleChrome/lighthouse/pull/12280

**Review by adamraine — APPROVED** [review]
